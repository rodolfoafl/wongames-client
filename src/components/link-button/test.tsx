import { render, screen } from '@testing-library/react'

import LinkButton from '.'

describe('<LinkButton />', () => {
  it('should render the medium size link button by default', () => {
    const { container } = render(<LinkButton href="/">Buy Now</LinkButton>)

    expect(screen.getByRole('link', { name: /Buy Now/i })).toHaveClass(
      'h-10 px-4 text-sm'
    )

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render the small size link button', () => {
    render(
      <LinkButton href="/" size="small">
        Buy Now
      </LinkButton>
    )

    expect(screen.getByRole('link', { name: /Buy Now/i })).toHaveClass(
      'h-[1.875rem] px-[0.875rem] text-xs'
    )
  })

  it('should render the large size link button', () => {
    render(
      <LinkButton href="/" size="large">
        Buy Now
      </LinkButton>
    )

    expect(screen.getByRole('link', { name: /Buy Now/i })).toHaveClass(
      'h-[3.125rem] px-5 text-base'
    )
  })

  it('should render the full width link button', () => {
    render(
      <LinkButton href="/" fullWidth>
        Buy Now
      </LinkButton>
    )

    expect(screen.getByRole('link', { name: /Buy Now/i })).toHaveClass('w-full')
  })
})
