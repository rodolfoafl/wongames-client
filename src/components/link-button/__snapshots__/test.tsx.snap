// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LinkButton /> should render the medium size link button by default 1`] = `
<a
  class="flex items-center justify-center rounded-sm border-none bg-[linear-gradient(180deg,#FF5F5F_0%,#F062C0_100%)] w-fit cursor-pointer p-2 font-semibold text-white hover:bg-[linear-gradient(180deg,#E35565_0%,#D958A6_100%)] h-10 px-4 text-sm"
  href="/"
>
  Buy Now
</a>
`;
